packages/react-server-dom-webpack/src/ReactFlightWebpackNodeRegister.js

```diff
@@ -15,8 +15,27 @@ const Module = require('module');
 
 module.exports = function register() {
   const CLIENT_REFERENCE = Symbol.for('react.client.reference');
+  const SERVER_REFERENCE = Symbol.for('react.server.reference');
   const PROMISE_PROTOTYPE = Promise.prototype;
 
+  // Patch bind on the server to ensure that this creates another
+  // bound server reference with the additional arguments.
+  const originalBind = Function.prototype.bind;
+  /*eslint-disable no-extend-native */
+  Function.prototype.bind = (function bind(this: any, self: any) {
+    // $FlowFixMe[unsupported-syntax]
+    const newFn = originalBind.apply(this, arguments);
+    if (this.$$typeof === SERVER_REFERENCE) {
```

Comment:
Since you already control the functions you need the special behavior for, judging by the `$$typeof` presence, there's no need to override the global prototype to preserve it across `bind` calls. `bind` preserves the prototype chain.

```js
const SERVER_REFERENCE = Symbol.for('server');

const put$$On = (fn) => {
    const ServerFunctionProto = Object.create(Function.prototype);
    ServerFunctionProto.$$typeof = SERVER_REFERENCE;
    Object.setPrototypeOf(fn, ServerFunctionProto);
}
// a function
const prebind = (a) => {
    console.log(`hello ${a}!`);
};
// the moment in time you control what the function is
put$$On(prebind);

// bind called elsewhere
const newFunction = prebind.bind({}, 'world')

console.log(newFunction.$$typeof)
newFunction()
```
prints
```
Symbol(server)
hello world!
```

Reply:
Luckily this isn’t the real implementation. This is just a reference implementation and then frameworks/bundlers do it differently. This is just showing an example.

Also, keep in mind that this file patches all .js imports using a deprecated Node.js api. Are you not concerned about that? 😛

We could probably get away with defining it as non-enumerable prop on the object along with the rest. setPrototypeOf has some really unfortunate consequences in VMs and doesn’t really help anything.

The issue with these techniques is always that the way people access these in indirections aren’t always the same. Eg libraries do Function.prototype.bind.call(…) to be clever which wouldn’t be caught by this technique.

This feature is non-essential though so probably ok.